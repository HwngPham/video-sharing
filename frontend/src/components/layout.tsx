import { ReactNode } from "react";
import { Navigation } from "./navigation";

export interface LayoutProps {
  children: ReactNode;
}
export const Layout = ({ children }: LayoutProps) => {
  return (
    <div>
      <Navigation />

      {children}
    </div>
  );
};
