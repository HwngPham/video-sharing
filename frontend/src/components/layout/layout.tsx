import { ReactNode } from "react";
import { Navigation } from "../navigation/navigation";

export interface LayoutProps {
  children: ReactNode;
}
export const Layout = ({ children }: LayoutProps) => {
  return (
    <div>
      <Navigation />
      <main>{children}</main>
    </div>
  );
};
