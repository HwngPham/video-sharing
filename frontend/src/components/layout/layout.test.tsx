import { render } from "@testing-library/react";
import { Layout } from "./layout";

describe("<Button />", () => {
  it("matches snapshot", () => {
    const container = render(<Layout>content</Layout>);
    expect(container).toMatchSnapshot();
  });
});
