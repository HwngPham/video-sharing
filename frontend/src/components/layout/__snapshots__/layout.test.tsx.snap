// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="fixed w-full bg-blue-500 p-6 rounded-sm"
        >
          <div
            class="container flex justify-between mx-8 flex-wrap mx-auto"
          >
            <h1
              class="text-xl text-white"
            >
              Funny Movies
            </h1>
            <div>
              <form
                class="text-sm flex gap-2 flex-wrap"
              >
                <div
                  class="flex flex-col gap-1 md:flex-row"
                >
                  <input
                    class="bg-white py-1 px-4 block w-full appearance-none leading-normal border border-gray-300 rounded-sm focus:outline-none focus:shadow-outline"
                    name="email"
                    placeholder="email"
                    required=""
                    value=""
                  />
                  <input
                    class="bg-white py-1 px-4 block w-full appearance-none leading-normal border border-gray-300 rounded-sm focus:outline-none focus:shadow-outline"
                    name="password"
                    placeholder="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
                  type="submit"
                >
                  Login / Register
                </button>
              </form>
            </div>
          </div>
        </nav>
        <main
          class="pt-24 container flex justify-between mx-8 flex-wrap mx-auto"
        >
          content
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="fixed w-full bg-blue-500 p-6 rounded-sm"
      >
        <div
          class="container flex justify-between mx-8 flex-wrap mx-auto"
        >
          <h1
            class="text-xl text-white"
          >
            Funny Movies
          </h1>
          <div>
            <form
              class="text-sm flex gap-2 flex-wrap"
            >
              <div
                class="flex flex-col gap-1 md:flex-row"
              >
                <input
                  class="bg-white py-1 px-4 block w-full appearance-none leading-normal border border-gray-300 rounded-sm focus:outline-none focus:shadow-outline"
                  name="email"
                  placeholder="email"
                  required=""
                  value=""
                />
                <input
                  class="bg-white py-1 px-4 block w-full appearance-none leading-normal border border-gray-300 rounded-sm focus:outline-none focus:shadow-outline"
                  name="password"
                  placeholder="password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
              <button
                class="bg-blue-500 text-white py-1 px-4 rounded-sm border cursor-pointer whitespace-nowrap hover:bg-blue-700 hover:border-transparent"
                type="submit"
              >
                Login / Register
              </button>
            </form>
          </div>
        </div>
      </nav>
      <main
        class="pt-24 container flex justify-between mx-8 flex-wrap mx-auto"
      >
        content
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
